contracts/ExitHandler.sol

```diff
@@ -47,6 +47,161 @@ contract ExitHandler is DepositHandler {
    */
   mapping(bytes32 => Exit) public exits;
 
+  struct Verification {
+    uint32 startTime;
+    uint224 stake;
+  }
+
+  event VerificationStarted(bytes32 indexed txHash);
+  /**
+   * txHash -> Verification Start Time mapping.
+   * Used to verify consolidate and computation transactions
+   * before they can be used to challenge exits.
+   */
+  mapping(bytes32 => Verification) public verifications;
+
+  /**
+   * Consolidate tx inputs are unsigned, hence an operator
+   * could create and include invalid consolidate txns to:
+   * a) challenge valid exits.
+   * b) decrease the priority of funds in exit queue by consolidating them,
+   *    putting the priority lower than his own transaction in the latest block.
+   *
+   * The following invalid txns are possible:
+   * - consolidates with out-of-no-where inputs. (a,b)
+   * - consolidates with inputs from utxos of other owners. (a,b)
+   * - consolidates with inputs from spent utxos of same owner. (a)
+   * - consolidates with duplicate inputs. (a)
+   *
+   * A verification process is introduced with half the duration of the exit procedure.
+   * During this verification a transaction can be challenged before it is whitelisted
+   * to be used in an exit challenge. We use this verification process to scrutinize 
+   * the validity of complex transactions (consolidate and spending conditions).
+   */
+  function startVerification(bytes32[] _proof, uint256 _outputIndex) public payable {
+    require(msg.value >= exitStake, "Not enough ether sent to pay for verification stake");
+    bytes32 parent;
+    (parent,,,) = bridge.periods(_proof[0]);
+    require(parent > 0, "The referenced period was not submitted to bridge");
+
+    // check exiting tx inclusion in the root chain block
+    bytes32 utxoId;
+    bytes memory txData;
+    (, utxoId, txData) = TxLib.validateProof(32, _proof);
+    if (_outputIndex > 0) {
+      utxoId = bytes32((_outputIndex << 248) | uint248(utxoId));
```

Comment:
i've tried `mapping(TxLib.Outpoint => Verification) public verifications;` before. but then i get this error from solidity:
```
Error parsing /Users/johann/dev/leap-contracts/contracts/ExitHandler.sol: ParsedContract.sol:61:11: ParserError: Expected elementary type name for mapping key type
```
i don't know how to achieve an elementary type without merging the `hash` and `pos`. because i know that `outputIndex < 16`, i wanted to truncate the hash as little as possible, to keep collision resistance as high as possible.